Add project status section to README.md

lachie83 opened the pull request:
As discussion during the weekly project meeting on Friday, June 25 2021. This adds a project status section to make it more clear about the project maturity.

cc @mnm678 @gokarnm

Signed-off-by: Lachlan Evenson <lachlan.evenson@microsoft.com>

lachie83 committed:
Add project status section to README.md

Signed-off-by: Lachlan Evenson <lachlan.evenson@microsoft.com>

mnm678 commented on README.md:
```diff
@@ -2,6 +2,10 @@
 
+## Project Status
+
+The Notary v2 project is in early development. Please refer to the [milestones](https://github.com/notaryproject/notaryproject/milestones) or [attend](#contributing-&-conversations) the weekly meetings for details on the roadmap.
```
```suggestion
The Notary v2 project is in early development and design documents should not be considered final. Please refer to the [milestones](https://github.com/notaryproject/notaryproject/milestones) or [attend](#contributing-&-conversations) the weekly meetings for details on the roadmap.
```

mnm678 replied:
I also left comments on some of the project milestones. The first one was targeted to be done this month, and as we are still working on requirements I think it should be pushed to the end of the year at the earliest.

lachie83 committed:
Update README.md
Update verbiage to include design documents and how they should be considered.

Co-authored-by: Marina Moore <mnm678@users.noreply.github.com>

Signed-off-by: Lachlan Evenson <lachlan.evenson@microsoft.com>

SteveLasker reviewed (approved):
LGTM
